billy-portugal-ebean/src/main/java/com/premiumminds/billy/portugal/persistence/dao/ebean/DAOPTCreditNoteEntryImpl.java

```diff
@@ -40,23 +45,20 @@ public PTCreditNoteEntryEntity getEntityInstance() {
 
     @Override
     public PTCreditNoteEntity checkCreditNote(PTInvoice invoice) {
+        List<JPAPTCreditNoteEntity> creditNotes = this.queryCreditNote().findList();
 
-        /*QJPAPTCreditNoteEntity creditNoteEntity = QJPAPTCreditNoteEntity.jPAPTCreditNoteEntity;
-        
-        JPAQuery query = new JPAQuery(this.getEntityManager());
-        
-        query.from(creditNoteEntity);
-        
-        List<JPAPTCreditNoteEntity> allCns = query.list(creditNoteEntity);
-        
         // TODO make a query to do this
-        for (JPAPTCreditNoteEntity cne : allCns) {
-            for (PTCreditNoteEntry cnee : cne.getEntries()) {
-                if (cnee.getReference().getNumber().compareTo(invoice.getNumber()) == 0) {
-                    return cne;
+        for (JPAPTCreditNoteEntity creditNote : creditNotes) {
+            for (PTCreditNoteEntry entry : creditNote.getEntries()) {
```

Comment:
I tried to translate this to Ebean query beans, but due to its inheritance limitations, it's even harder to do so than translating to hibernate queries...

This is terrible indeed. It is a good example of how extremely valuable bidirectional relations can be. The root of the problem current is that the relation between a CreditNoteEntry and its referenced PTInvoice is unidirectional. If it were bidirectional, one could simply obtain only the CreditNoteEntries referenced by a certain PTInvoice; not a single query needed.

Anyway, I believe that it would be time-consuming to transform this code, and at this point in time there are other priorities.

Reply:
Add a TODO please, in order to address this later.

Reply:
The code already had a TODO. I created an issue for this: https://github.com/premium-minds/billy/issues/25